src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1001,27 +1001,27 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     String current_database = getContext()->getCurrentDatabase();
     auto database_name = create.database ? create.getDatabase() : current_database;
 
+    DDLGuardPtr ddl_guard;
```

Comment:
you can move this to line 1023

Reply:
No, I cannot, because this lock must be held after the scope between lines 1008 and 1050 ends

Reply:
Ah got confused with indentations.